### PR TITLE
Correct `elite_c` converter

### PR DESCRIPTION
**Pull request by dunk2k**
## Description

Relates to #20330 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* n/a

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by dunk2k** [comment]
Is it safe to assume that all keyboards with
`"processor": "atmega32u4"` and `"bootloader": "atmel-dfu"` are an Elite-C micro controller unit?

File: keyboards/lily58/r2g/info.json
Comment (waffle87) [inline keyboards/lily58/r2g/info.json:1]: This keyboard does not use an Elite-C.

**Comment by waffle87** [comment]
Probably something like 10% of keyboards with `atmel-dfu`/`atmega32u4` settings use an Elite-C. I suggest you carefully confirm each of the keyboards you're changing are meant to be built with an Elite-C, and are not using an integrated microcontroller.

File: keyboards/lily58/r2g/info.json
Comment (dunk2k) [reply keyboards/lily58/r2g/info.json:1]: `master` branch shows, for named keyboard:
```
"processor": "atmega32u4",
"bootloader": "atmel-dfu",
```
More than happy to change these details to confirmed controller.

Comment (dunk2k) [reply keyboards/lily58/r2g/info.json:1]: reverted

**Comment by zvecr** [comment]
> Correct elite_c converter

None of this actually "corrects" the converters.

The changes to `data/mappings/defaults.hjson` along with the keyboard changes are not something I would want to accept at the moment, as it will break valid conversions where the board isnt actually using the extra elite c pins.

Closing as the rest of the changes are something that is already being handled elsewhere.

**Pull request closed without merging**